instrumentor/internal/webhook_env_injector/webhook_env_injector.go

```diff
@@ -0,0 +1,168 @@
+		return
+	}
+
+	for _, envVarName := range envVarsPerLanguage {
+		if handleManifestEnvVar(container, envVarName, otelsdk, logger) {
+			continue
+		}
+
+		err := injectEnvVarsFromRuntime(ctx, p, logger, podWorkload, container, envVarName, otelsdk)
+		if err != nil {
+			logger.Error(err, "failed to inject environment variables for container", "container", container.Name)
+		}
+	}
+}
+
+func getEnvVarsForLanguage(pl common.ProgrammingLanguage) []string {
+	// Check if the key exists in the map - for safety
+	if envVars, exists := envOverwrite.EnvVarsForLanguage[pl]; exists {
+		return envVars
+	}
+
+	// Return nil if the key doesn't exist
+	return nil
+}
+
+func handleManifestEnvVar(container *corev1.Container, envVarName string, otelsdk common.OtelSdk, logger logr.Logger) bool {
+	manifestEnvVar := getContainerEnvVarPointer(&container.Env, envVarName)
+	if manifestEnvVar == nil {
+		return false // Not found in manifest. further process it
+	}
+
+	possibleValues := envOverwrite.GetPossibleValuesPerEnv(manifestEnvVar.Name)
+	if possibleValues == nil {
+		return true // Skip further processing
+	}
+
+	odigosValueForOtelSdk := possibleValues[otelsdk]
+	if strings.Contains(manifestEnvVar.Value, "/var/odigos/") {
+		logger.Info("env var exists in the manifest and already includes odigos values", "envVarName", envVarName)
+		return true // Skip further processing
+	}
+
+	updatedEnvValue := updatedGetPatchedEnvValue(envVarName, manifestEnvVar.Value, odigosValueForOtelSdk)
+	if updatedEnvValue != nil {
+		manifestEnvVar.Value = *updatedEnvValue
+		logger.Info("updated manifest environment variable", "envVarName", envVarName, "value", *updatedEnvValue)
+	}
+	return true // Handled, no need for further processing
+}
+
+func injectEnvVarsFromRuntime(ctx context.Context, p client.Client, logger logr.Logger, podWorkload workload.PodWorkload,
+	container *corev1.Container, envVarName string, otelsdk common.OtelSdk) error {
+
+	var workloadInstrumentationConfig v1alpha1.InstrumentationConfig
+	instrumentationConfigName := fmt.Sprintf("%s-%s", strings.ToLower(string(podWorkload.Kind)), podWorkload.Name)
+
+	if err := p.Get(ctx, client.ObjectKey{Namespace: podWorkload.Namespace, Name: instrumentationConfigName}, &workloadInstrumentationConfig); err != nil {
+		return fmt.Errorf("failed to get instrumentationConfig: %w", err)
+	}
+
+	runtimeDetails := workloadInstrumentationConfig.Status.GetRuntimeDetailsForContainer(*container)
+	if runtimeDetails == nil {
+		logger.Error(nil, "failed to get runtime details for container", "container", container.Name)
+		return nil
+	}
+
+	if !shouldInject(runtimeDetails, logger, container.Name) {
+		return nil
+	}
+
+	envVarsToInject := prepareEnvVars(runtimeDetails, envVarName, otelsdk)
+	container.Env = append(container.Env, envVarsToInject...)
+	return nil
+}
+
+func prepareEnvVars(runtimeDetails *v1alpha1.RuntimeDetailsByContainer, envVarName string, otelsdk common.OtelSdk) []corev1.EnvVar {
```

Comment:
there are so many env vars involved (manifest, envOverwriter, runtime detection). what does it mean "prepare"?

From it's name, it is impossible to understand it's role.